spark2.4/src/main/scala/MessageCounter.scala

```diff
@@ -0,0 +1,345 @@
+      process
+    } catch {
+      case e: Exception => {
+        val msg = "MessageCounter: Problem processing." +
+          "\n" +
+          e.toString
+        println(msg)
+        throw e
+      }
+      case t: Throwable => {
+        val msg = "MessageCounter: Problem processing." +
+          "\n" +
+          t.toString
+        println(msg)
+        throw t
+      }
+    } finally {
+      close
+    }
+  }
+  
+//  def init(): Unit = {
+//    var kafkaReady = false
+//    var dbReady = false
+//    while( (!kafkaReady) || (!dbReady) ) {
+//      kafkaTotal = kafkaCount
+//      dbTotal = dbCount
+//      if( (!kafkaReady) && kafkaTotal > 0 ) {
+//        kafkaStart = now
+//        kafkaReady = true
+//      }
+//      if( (!dbReady) && dbTotal > 0 ) {
+//        dbStart = now
+//        dbReady = true
+//      }
+//      Thread.sleep(1000)
+//    }
+//  }
+
+  def now(): LocalDateTime = LocalDateTime.now
+  var kt0: LocalDateTime = _
+  var kt1: LocalDateTime = _
+  var dt0: LocalDateTime = _
+  var dt1: LocalDateTime = _
+  var t0: LocalDateTime = _
+  
+  def process(): Unit = {
+    var prevKcount = 0L
+//    do {
+//      Thread.sleep(500)
+//      prevKcount = kafkaCount
+//    } while( prevKcount == 0L )
+    
+    t0 = now
+    kt0 = now
+    dt0 = now
+    var count = 0L
+    displayHeader
+    while(true) {
+//      println(s"\n${(new java.util.Date).toString} Check DB")
+      count = dbCount
+      dt1 = now
+      dbTotal += count
+      dbRate = count.asInstanceOf[Double] / SECONDS.between(dt0, dt1)
+      dt0 = dt1
+
+//      println(s"${(new java.util.Date).toString} Check Kafka")
+      count = kafkaCount
+      kt1 = now
+      kafkaTotal = count
+      kafkaRate = (count-prevKcount).asInstanceOf[Double] / SECONDS.between(kt0, kt1)
+      kt0 = kt1
+      prevKcount = count
+
+      display
+      Thread.sleep(1000)
+    }
+  }
+
+  def displayHeader(): Unit = {
+    println(" ")
+    //println(s"${(new java.util.Date).toString}")
+    println("Tm(s)\tIn Ttl\tIn Rt\tIn Rt Min5\tIn Rt Avg5\tIn Rt Max5\tDB Ttl\tDB Rt\tDB Rt Min5\tDB Rt Avg5\tDB Rt Max5\tLag")
+    //    println("\t\tAvg In Rate /s\t\tAvg Table Rate /s")
+  }
+  
+  def display(): Unit = {
+//    println(" ")
+//    println("****************")
+//    println((new java.util.Date).toString)
+//    println(" ")
+//    println("Incoming Total")
+//    println( kafkaTotal )
+//    println("Incoming Rate")
+//    println( kafkaTotal.asInstanceOf[Double] / SECONDS.between(kafkaStart, now) )
+//    println(" ")
+//    println("In Table Total")
+//    println( dbTotal )
+//    println("In Table Rate")
+//    println( dbTotal.asInstanceOf[Double] / SECONDS.between(dbStart, now) )
+//    println(" ")
+//    println("Records in Process")
+//    println( kafkaTotal - dbTotal )
+    
+//    val kafkaRate = kafkaTotal.asInstanceOf[Double] / SECONDS.between(kafkaStart, now)
+//    val dbRate = dbTotal.asInstanceOf[Double] / SECONDS.between(dbStart, now)
```

Comment:
Instead of having commented out code like this, another option could be to define a boolean variable controlling whether that section of code is executed or not.  The name of the boolean could describe what the code block does.  Even better would be to add a function argument to control it.